packages/alfa-style/src/property/font.ts

```diff
@@ -111,24 +118,47 @@ export namespace Font {
    */
   export const Size: Property<Size.Specified, Size.Computed> = Property.of(
     Length.of(16, "px"),
-    either(
-      either(
-        Keyword.parse(
-          "xx-small",
-          "x-small",
-          "small",
-          "medium",
-          "large",
-          "x-large",
-          "xx-large",
-          "xxx-large"
-        ),
-        Keyword.parse("larger", "smaller")
+    either<Slice<Token>, Size.Specified, string>(
+      Keyword.parse(
+        "xx-small",
+        "x-small",
+        "small",
+        "medium",
+        "large",
+        "x-large",
+        "xx-large",
+        "xxx-large"
       ),
-      either(Percentage.parse, Length.parse)
+      Keyword.parse("larger", "smaller"),
+      Percentage.parse,
+      Length.parse,
+      filter(
+        Calculation.parse,
+        ({ expression: { kind } }) =>
+          kind.is("length", 1, true) || kind.is("percentage"),
+        () => `calc() expression must be of type "length" or "percentage"`
+      )
     ),
     (style) =>
       style.specified("font-size").map((size) => {
+        if (size.type === "calculation") {
+          const { expression } = size.reduce((value) => {
+            if (Length.isLength(value)) {
+              return Resolver.length(value, style.parent);
+            }
+
+            if (Percentage.isPercentage(value)) {
+              const parent = style.parent.computed("font-size").value;
+
+              return Length.of(parent.value * value.value, parent.unit);
+            }
+
+            return value;
+          });
+
+          size = expression.toLength().or(expression.toPercentage()).get();
+        }
+
```

Comment:
Shouldn't the `calc` evaluation happen at a higher level (`compute`, I guess)?
As far as I understand, `calc` could happen almost anywhere and we probably don't want to handle the evaluation at property level… It's `Length` (and others) that should allow `Calculation` as values.

Reply:
Resolution of `calc()` expressions is specific to each property that accepts them and so is the way `calc()` expressions are parsed. For example, the `font-size` property only allows `calc()` expressions of type `<length-percentage>` and so `calc()` expressions with a different type should be rejected at parse time. At computed value time, the `font-size` property knows to resolve percentages in `calc()` expressions against the parent `font-size`, but this is different from property to property. For even other properties, such as `width` and `height`, `calc()` expressions can't even be fully resolved at computed value time and will carry over.

So, it's... complicated 🙈